core/src/main/java/bio/terra/pearl/core/service/participant/EnrolleeRelationService.java

```diff
@@ -51,6 +42,19 @@ public Enrollee newGovernedEnrolleeCreationRecord(Enrollee enrollee, Participant
                 .build();
 
         this.create(enrolleeRelation);
-        return  enrollee;
+        return enrollee;
+    }
+
+    @Override
+    @Transactional
+    public void delete(UUID enrolleeRelationId, Set<CascadeProperty> cascades) {
+        dao.delete(enrolleeRelationId);
+    }
+
+    public void deleteByEnrolleeId(UUID enrolleeId) {
+        List<EnrolleeRelation> enrolleeRelations = dao.findByEnrolleeId(enrolleeId);
+        for (EnrolleeRelation enrolleeRelation : enrolleeRelations) {
+            dao.delete(enrolleeRelation.getId());
+        }
```

Comment:
This could be simplified to one query by adding a `deleteByEnrolleeId` method to the `EnrolleeRelationDao`, which would call `deleteByProperty("enrollee_id", enrolleeId)`

Reply:
True, although once we start attaching stuff to EnrolleeRelations (surveyResponses, etc...) we'll need to do the deletes individually anyway so that we can clear out all those things first

Reply:
Yes I don't expect delete to stay like this for long. Right now the only real functionality of it is in the populate classes, but eventually if we want to have that delete ability for the users and throughout the system then we need to dig much deeper into what delete vs withdrawal means.